R/sha1.R

```diff
@@ -398,30 +398,35 @@ num2hex <- function(x, digits = 14L, zapsmall = 7L){
     output <- rep(NA_character_, length(x))
     output[x.inf & x > 0] <- "Inf"
     output[x.inf & x < 0] <- "-Inf"
-    x.finite <- !x.na & !x.inf
-    x.hex <- sprintf("%a", x[x.finite])
-    exponent <- as.integer(gsub("^.*p", "", x.hex))
     # detect "small" numbers
-    zapsmall.hex <- floor(log2(10 ^ -zapsmall))
-    zero <- x.hex == sprintf("%a", 0) | exponent <= zapsmall.hex
-    if (any(zero)) {
-        output[x.finite][zero] <- "0"
-        if (all(zero)) {
-            return(output)
-        }
-    }
+    x.zero <- !x.na & !x.inf & abs(x) <= (2^floor(log2(10 ^ -zapsmall)))
+    if (any(x.zero)) {
```

Comment:
I think we can even drop this `ìf` statement as we don't have it in lines 399 and 400. We just need to make sure that is works as expected when `x.zero` contains only `FALSE` values.

Reply:
I just verified with

```r
> output <- "1"
> x.zero <- FALSE
> output[x.zero] <- "0"
```

which worked as expected.